frontend/src/pushapi.js

```diff
@@ -5,93 +5,42 @@ angular.module('clouway-push', [])
 
   .provider('pushApi', function () {
 
-    this.subscriberLength = 15;
-
     this.endpoints = {
       serviceUrl: ""
     };
 
-    this.timeIntervals = {
-      keepAlive: 60,
-      channelReconnect: 5
-    };
-
     /**
      * Sets the backend url connection.
      *
      * @param url the new url
      * @returns {*} the provider instance for chaining purposes.
      */
-    this.backendServiceUrl = function(url) {
+    this.backendServiceUrl = function (url) {
       this.endpoints.serviceUrl = url;
       return this;
     };
 
-
-    /**
-     * Set a time interval for sending a keepAlive.
-     *
-     * @param {number} seconds time in seconds between each keepAlive.
-     * @returns {*} the provider instance for chaining purposes.
-     */
-    this.keepAliveTimeInterval = function (seconds) {
-      this.timeIntervals.keepAlive = seconds;
-      return this;
-    };
-
-    /**
-     * Set a time interval for attempting a reconnect.
-     *
-     * @param {number} seconds time in seconds between each reconnect attempt.
-     * @returns {*} the provider instance for chaining purposes.
-     */
-    this.reconnectTimeInterval = function (seconds) {
-      this.timeIntervals.channelReconnect = seconds;
-      return this;
-    };
-
-    this.$get = function ($rootScope, $interval, $timeout, $window, $http) {
-      var subscriberLength = this.subscriberLength;
-      var timeIntervals = this.timeIntervals;
+    this.$get = function ($rootScope, $interval, $timeout, $window, $http, $q) {
       var endpoints = this.endpoints;
       var boundEvents = {};
       var pendingBulkBindings = [];
       var pendingInitialBindings = [];
-      var channelSubscriber;
-      var keepAliveInterval;
 
       var initialFlushTimeout;
 
       var service = {};
-
-      /**
-       * Generate a random alpha-numeric subscriber.
-       *
-       * @returns {string} the generated subscriber.
-       */
-      var generateSubscriber = function (length) {
-        var symbols = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ123456';
-        var generated = [];
-        var i = 0;
-
-        while (i < length) {
-          i++;
-          var character = Math.floor(Math.random() * symbols.length);
-          generated.push(symbols.charAt(character));
-        }
-
-        return generated.join('');
-      };
+      var firebaseDatabaseRef = {};
+      var customerName = '';
 
       /**
        * Handle push event message.
        *
-       * @param {Object} message
+       * @param {Object} data
```

Comment:
The parameter is now of type `{String}`.

Reply:
Fixed. Now comment is ```* @param {String} message```